temporalio/contrib/openai_agents/open_ai_data_converter.py

```diff
@@ -0,0 +1,155 @@
+
+
+class _WrapperModel(RootModel[T]):
+    model_config = {
+        "arbitrary_types_allowed": True,
+    }
+
+
+class _OpenAIJSONPlainPayloadConverter(EncodingPayloadConverter):
+    """Payload converter for OpenAI agent types that supports Pydantic models and standard Python types.
+
+    This converter extends the standard JSON payload converter to handle OpenAI agent-specific
+    types, particularly Pydantic models. It supports:
+
+    1. All Pydantic models and their nested structures
+    2. Standard JSON-serializable types
+    3. Python standard library types like:
+       - dataclasses
+       - datetime objects
+       - sets
+       - UUIDs
+    4. Custom types composed of any of the above
+
+    The converter uses Pydantic's serialization capabilities to ensure proper handling
+    of complex types while maintaining compatibility with Temporal's payload system.
+
+    See https://docs.pydantic.dev/latest/api/standard_library_types/ for details
+    on supported types.
+    """
+
+    @property
+    def encoding(self) -> str:
+        """Get the encoding identifier for this converter.
+
+        Returns:
+            The string "json/plain" indicating this is a plain JSON converter.
+        """
+        return "json/plain"
+
+    def to_payload(self, value: Any) -> Optional[temporalio.api.common.v1.Payload]:
+        """Convert a value to a Temporal payload.
+
+        This method wraps the value in a Pydantic RootModel to handle arbitrary types
+        and serializes it to JSON.
+
+        Args:
+            value: The value to convert to a payload.
+
+        Returns:
+            A Temporal payload containing the serialized value, or None if the value
+            cannot be converted.
+        """
+        wrapper = _WrapperModel[Any](root=value)
+        data = wrapper.model_dump_json(exclude_unset=True).encode()
+
+        return temporalio.api.common.v1.Payload(
+            metadata={"encoding": self.encoding.encode()}, data=data
+        )
+
+    def from_payload(
+        self,
+        payload: temporalio.api.common.v1.Payload,
+        type_hint: Optional[Type] = None,
+    ) -> Any:
+        """Convert a Temporal payload back to a Python value.
+
+        This method deserializes the JSON payload and validates it against the
+        provided type hint using Pydantic's validation system.
+
+        Args:
+            payload: The Temporal payload to convert.
+            type_hint: Optional type hint for validation.
+
+        Returns:
+            The deserialized and validated value.
+
+        Note:
+            The type hint is used for validation but the actual type returned
+            may be a Pydantic model instance.
+        """
+        _type_hint = type_hint if type_hint is not None else Any
+        wrapper = _WrapperModel[_type_hint]  # type: ignore[valid-type]
+        # Needed due to
+        # if TYPE_CHECKING:
+        #     from .agent import Agent
+        #
+        # in the agents/items.py
```

Comment:
I don't think the comment makes sense; `TYPE_CHECKING` is `False` at run-time.


https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING